speaker/main.go

```diff
@@ -277,7 +282,7 @@ func (c *controller) SetBalancer(l log.Logger, name string, svc *v1.Service, eps
 		}
 	}
 
-	if svcIP, ok := c.svcIP[name]; ok && !lbIP.Equal(svcIP) {
+	if svcIPs, ok := c.svcIPs[name]; ok && !reflect.DeepEqual(lbIPs, svcIPs) {
```

Comment:
`ip.Equal` does a bit more than bit comparison: https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/net/ip.go;l=432
I don't have a strong opinino, but I think using it would put us on the safe side (considering the old implementation was doing the same). That would mean writing a custom comparison functoin for two []IP

Reply:
added customer compare